### PR TITLE
Vault: Adding ability to open cluster ip in k8s service

### PR DESCRIPTION
**Pull request by CharlieC3**
#### What this PR does / why we need it:
This PR adds a couple new values and updates the k8s Vault service to open up the cluster IP for high availability communication.

More specifically, this PR helps resolve this error in Vault logs:
```
2018-09-28T13:35:59.370Z [ERROR] core: handleRequestForwarding: error forwarding request: error="error during forwarding RPC request"
2018-09-28T13:44:10.912Z [ERROR] core: error during forwarded RPC request: error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 172.20.142.28:8201: i/o timeout""
```

#### Which issue this PR fixes
None

### CODE REVIEW

**Comment by CharlieC3** [comment]
@linki @viglesiasce 
Any updates on reviewing this PR? I'd like to have this merged in soon if possible. Thanks!

**Comment by CharlieC3** [comment]
@linki @viglesiace
Are there any issues with this pr you'd like addressed?

File: incubator/vault/templates/service.yaml [hunk 1]
```diff
@@ -27,6 +27,10 @@ spec:
     protocol: TCP
     targetPort: {{ .Values.service.port }}
     name: api
+  - port: {{ .Values.service.clusterExternalPort }}
```
Comment (viglesiasce) [inline incubator/vault/templates/service.yaml:34]: Should this port always be exposed or should it be flagged behind an if block?

File: incubator/vault/Chart.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
 description: A Helm chart for Vault, a tool for managing secrets
 name: vault
-version: 0.14.1
-appVersion: 0.11.2
+version: 0.14.2
+appVersion: 0.11.3
```
Comment (viglesiasce) [inline incubator/vault/Chart.yaml]: The app version didn't change in this case AFAIK.
```suggestion
appVersion: 0.11.2
```

**Review by viglesiasce — CHANGES_REQUESTED** [review]

File: incubator/vault/templates/service.yaml (hunk 1 above)
Comment (CharlieC3) [reply incubator/vault/templates/service.yaml:34]: I added a conditional block to preserve the original design of this service. When a value for the `service.clusterExternalPort` is provided, the cluster port will be exposed in the service.

**Comment by CharlieC3** [comment]
@viglesiasce I've addressed your concerns above, let me know if there's anything else!

**Comment by CharlieC3** [comment]
@linki
@viglesiasce
Can this PR be reopened? This issue is still appearing in Vault deployments using this chart and it's a major issue.

See references below:
https://github.com/hashicorp/vault/issues/3885
https://stackoverflow.com/questions/48321559/http-handlerequestforwarding-error-forwarding-request-error-error-during-forwa

**Comment by CharlieC3** [comment]
@linki
@viglesiasce
Please comment

**Review by viglesiasce — APPROVED** [review]

**Comment by CharlieC3** [comment]
Thanks @viglesiasce !

**Pull request merged**